src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -281,16 +282,16 @@ private static void OnRequestSendingRequest(WinHttpRequestState state)
                         out chain,
                         out sslPolicyErrors);
 
-                    bool result = state.ServerCertificateValidationCallback(
+                    result = state.ServerCertificateValidationCallback(
                         state.RequestMessage,
                         serverCertificate,
                         chain,
                         sslPolicyErrors);
-                    if (!result)
-                    {
-                        throw WinHttpException.CreateExceptionUsingError(
-                            (int)Interop.WinHttp.ERROR_WINHTTP_SECURE_FAILURE);
-                    }
+                }
+                catch (Exception ex)
+                {
+                    throw WinHttpException.CreateExceptionUsingError(
+                        (int)Interop.WinHttp.ERROR_WINHTTP_SECURE_FAILURE, ex);
```

Comment:
The description of ERROR_WINHTTP_SECURE_FAILURE is "One or more errors were found in the Secure Sockets Layer (SSL) certificate sent by the server."... is that the right error code to be using when it's the caller-supplied callback that's throwing?

Reply:
There is no other WinHttp specific error to return here and I didn't want to invent a new one. From WinHttp's point of view, the validation failed due to "errors" about the certificate validation process.  So, it's not perfect but we use it.  

FWIW, WinHttp returns this error even if there is nothing wrong with the certificate itself, but rather the SSL handshake has a problem (cipher suites mismatch, TLS protocol version mismatch etc.).  So, the error number is really right....it's just that the error message string from WinHttp.dll isn't as descriptive as it should be.